src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -5,20 +5,117 @@
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+
+    private Entry<K, V>[] bucketsTable;
+    private int size;
+    private int newCapacity;
+
+    public MyHashMap() {
+        size = 0;
+        newCapacity = 16;
+        bucketsTable = new Entry[INITIAL_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
+        double threshold = (double)size / bucketsTable.length;
+        if (threshold >= DEFAULT_LOAD_FACTOR) {
+            resizeTable();
+        }
+        if (key == null) {
+            putNullKey(value);
+            return;
+        }
+        int index = key == null
+                ? 0 : getHash(key);
+        if (bucketsTable[index] == null) {
+            bucketsTable[index] = new Entry<>(key, value, null);
+            size++;
+            return;
+        }
+        Entry<K, V> newEntry = bucketsTable[index];
+        while (newEntry != null) {
+            if (key.equals(newEntry.key)) {
+                newEntry.value = value;
+                return;
+            }
+            newEntry = newEntry.nextEntry;
+        }
+        bucketsTable[index] = new Entry<>(key, value, bucketsTable[index]);
+        size++;
+    }
 
+    private void putNullKey(V value) {
+        Entry<K, V> newEntry = bucketsTable[0];
```

Comment:
We are going to put all Entries with key == null to zero bucket. Let's modify getHash() so it will return 0 if key == null and then we can remove this method